File: Anushka-Pachauri/SlidingWindow/GetEqualSubstringsWithinBudget.java
class GetEqualSubstringsWithinBudget {
    public int equalSubstring(String s, String t, int maxCost) {
        int start = 0;
        int end = 0;
        int maxLength = 0;
        char[] s1 = s.toCharArray();
        char[] t1 = t.toCharArray();
        while (end < s1.length) {
            maxCost -= Math.abs(s1[end] - t1[end]);
            while (maxCost<0) {
                maxCost += Math.abs(s1[start] - t1[start]);
                start++;
            }
            maxLength = Math.max(maxLength, end - start + 1);
            end++;
        }
        return maxLength;
    }
}

File: Lakshay Garg/Leetcode/Sliding Window/Get Equal Substrings Within Budget.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int m) {
        int start=0,end=0,ml=0,cost=0;
        while(end<s.size()){
            cost+=abs(s[end]-t[end]);
            while(cost>m){
                cost-=abs(s[start]-t[start]);
                start++;
            }
            ml=max((1+end-start),ml);
            end++;
        }
        return ml;
    }
};
